### PR TITLE
feat: make pod dns configuration optional

### PR DESCRIPTION
**Pull request by jsievenpiper**
This change accomplishes two things:
 - Makes the pod DNS policy configurable (so those of us who don't want to run Google as an upstream do not have to)
 - Allows the pod DNS policy to be completely omitted. This allows the default policy to run, which helps compatibility with service meshes like istio to be put in place as well.

### CODE REVIEW

**Comment by jsievenpiper** [comment]
I have a fork of this that has released a `v2.1.0` if you'd like to test!

**Comment by jsievenpiper** [comment]
Looks like the cluster failed to come up, might need a new run?

**Comment by MoJo2600** [comment]
Hm... why is it not working

**Comment by MoJo2600** [comment]
Ok, I fixed the lint testing. Just one thing, could you please regenerate the documentation with helm-docs and extend it with the new settings? I'm happy to merge your code then.

**Comment by jsievenpiper** [comment]
Updated! @MoJo2600 thanks for the quick iteration with me!

File: charts/pihole/README.md [hunk 1]
```diff
@@ -249,7 +281,7 @@ MetalLB 0.7.3 has a bug, where the service is not announced anymore, when the po
 
 Feel free to contribute by making a [pull request](https://github.com/MoJo2600/pihole-kubernetes/pull/new/master).
 
-Please read [Contribution Guide](../../CONTRIBUTING.md) for more information on how you can contribute to this Chart.
+Please read [Contribution Guide](CONTRIBUTING.md) for more information on how you can contribute to this Chart.
```
Comment (MoJo2600) [inline charts/pihole/README.md:284]: The path is wrong in your commit, could you please set it to `../../Contributing.md`.

**Review by MoJo2600 — CHANGES_REQUESTED** [review]
Could you please change the link that is wrong?

File: charts/pihole/README.md (hunk 1 above)
Comment (jsievenpiper) [reply charts/pihole/README.md:284]: Yep happy to manually change this -- is that normal? This was the automatic output of `helm-docs`

**Review by MoJo2600 — APPROVED** [review]

**Pull request merged**